examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -1058,17 +1048,52 @@ def load_model_hook(models, input_dir):
                 raise ValueError(f"unexpected save model: {model.__class__}")
 
         lora_state_dict, network_alphas = LoraLoaderMixin.lora_state_dict(input_dir)
-        LoraLoaderMixin.load_lora_into_unet(lora_state_dict, network_alphas=network_alphas, unet=unet_)
 
-        text_encoder_state_dict = {k: v for k, v in lora_state_dict.items() if "text_encoder." in k}
-        LoraLoaderMixin.load_lora_into_text_encoder(
-            text_encoder_state_dict, network_alphas=network_alphas, text_encoder=text_encoder_one_
-        )
```

Comment:
We cannot be using `load_lora_into_unet()` and `load_lora_into_text_encoder()` for the following reason (described only for the UNet but applicable to the text encoders, too).  

* We call `add_adapter()` once on unet at the beginning of training. This creates an adapter config inside of the UNet.
* Then during loading an intermediate checkpoint in the accelerate hook, we call `load_lora_into_unet()`. It internally again calls `inject_adapter_in_model()` with the config inferred from the state dict provided. So, it internally creates another adapter. This is undesirable, right?